packages/frontend-app-api/src/tree/resolveAppNodeSpecs.ts

```diff
@@ -26,6 +30,8 @@ import {
 // eslint-disable-next-line @backstage/no-relative-monorepo-imports
 import { toInternalExtension } from '../../../frontend-plugin-api/src/wiring/resolveExtensionDefinition';
 
+export const rootPlugin = createFrontendPlugin({ pluginId: 'root' });
```

Comment:
This is the tricky bit, is it fine that this shows up in tests and as the plugin of the root extension?

We also considered detecting and using the `app` plugin if one is available, and otherwise fall back to a placeholder `app` plugin.

Kinda related, I'm thinking it's probably best to create this instance within `resolveAppNodeSpecs` rather than here, just to avoid the global state.

Reply:
Thinking a bit more about this I think my favorite solution for this is to default it to the `app` plugin but fall back to a new `app` plugin if it doesn't exist. I'm feeling that's gonna have the least impact on the overall structure, rather than introducing a new `root` plugin

Reply:
I just applied this change. The only thing that seems inconsistent to me is that the internal extension isn't returned when we try to get it by calling `getExtension` from the app plugin. But as we discussed, it might be better to remove this method from the node spec anyway.